### ISSUE TITLE
[Bug Report] libxml2 and libxslt are not installed by default for 22.04 docker requirements

### ISSUE THREAD

**Issue by JushBJJ** [tenstorrent/tt-metal#11859]
**Describe the bug**
For 22.04 Dockerfiles, libxml2 and libxslt are not installed by default, causing an error during `./create_venv` installation.

**To reproduce**
1. After building the docker images, clone the TT-Metal repository at root directory
2. Set environment variables
```sh
export ARCH_NAME=grayskull
export TT_METAL_HOME=/tt-metal
export PYTHON_PATH=/tt-metal
```
3. Run the following command
```sh
./create_venv
```

**Expected Behaviour**
Virtual environment should be succesfully created without any errors.

**Error**
```
root@eb3f60526a10:/tt-metal# ./create_venv.sh
Creating virtual env in: /opt/tt_metal_infra/tt-metal/python_env
...
Collecting lxml==4.9.4
  Downloading lxml-4.9.4.tar.gz (3.6 MB)
     |████████████████████████████████| 3.6 MB 8.4 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /opt/tt_metal_infra/tt-metal/python_env/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-6jh1ewwv/lxml/setup.py'"'"'; __file__='"'"'/tmp/pip-install-6jh1ewwv/lxml/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-itaf3txn
         cwd: /tmp/pip-install-6jh1ewwv/lxml/
    Complete output (3 lines):
    Building lxml version 4.9.4.
    Building without Cython.
    Error: Please make sure the libxml2 and libxslt development packages are installed.
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

For https://github.com/tenstorrent/tt-metal/issues/10469

**Comment by TT-billteng** [comment]
@tt-rkim did you add something that required these dependencies recently?

**Comment by tt-rkim** [comment]
Don't think so, could you send the full output so we can see where this requirement is coming from? @JushBJJ

**Comment by JushBJJ** [comment]
@tt-rkim
Here: https://pastebin.com/raw/Hjg2Q00X

Did a little bit of testing just now, changing the pip version to 22.0.2 in `./create_venv.sh` fixes the error while staying at pip v20.1.1  brings the error back. I originally suggested bumping the pip version of create_venv here https://github.com/tenstorrent/tt-metal/issues/11144#issuecomment-2307375188

**Comment by tt-rkim** [comment]
I see - it's weird that a lower version of pip causes you to require these system dependencies. One thread of work we could check out to make sure we have the full picture is to use `pipdeptree` on installing the virtual env with those system libraries installed. Not that important though.

I've posted a longer comment here explaining the problem: https://github.com/tenstorrent/tt-metal/issues/11144#issuecomment-2318184316

tl;dr: it's to support editable install workflow, fix to unblock you guys may be to just check for ubuntu version and use higher pip version if `>20.04`

**Comment by JushBJJ** [comment]
Should I remove libxml2-dev and libxslt-dev from the original PR for now and go forward with fixing #11859?

**Comment by TT-billteng** [comment]
> Should I remove libxml2-dev and libxslt-dev from the original PR for now and go forward with fixing #11859 only?

nah it's ok, we'll figure out why they're needed for 22.04 later

**Comment by TT-billteng** [comment]
> I see - it's weird that a lower version of pip causes you to require these system dependencies. One thread of work we could check out to make sure we have the full picture is to use `pipdeptree` on installing the virtual env with those system libraries installed. Not that important though.
> 
> I've posted a longer comment here explaining the problem: [#11144 (comment)](https://github.com/tenstorrent/tt-metal/issues/11144#issuecomment-2318184316)
> 
> tl;dr: it's to support editable install workflow, fix to unblock you guys may be to just check for ubuntu version and use higher pip version if `>20.04`

we should try cleaning out our pip dependencies, it's turning into a rat's nest

**Comment by TT-billteng** [comment]
closing as this is now part of the 22.04 docker dependencies
